include/bluetooth/mesh/vnd/dm_common.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file
+ * @defgroup bt_mesh_dm Distance Measurment Models
+ * @{
+ * @brief Common API for the Distance Measurment models.
+ */
+
+#ifndef BT_MESH_DM_H__
+#define BT_MESH_DM_H__
+
+#include <bluetooth/mesh/model_types.h>
+#include <dm.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define BT_MESH_VENDOR_COMPANY_ID 0x0059
+#define BT_MESH_MODEL_ID_DM_CLI 0x000B
+#define BT_MESH_MODEL_ID_DM_SRV 0x000C
+
+/** Parameters for the Distance measurement result entry. */
+struct bt_mesh_dm_res_entry {
+	/** Mode used for ranging. */
+	enum dm_ranging_mode method;
+	/** Quality indicator */
+	enum dm_quality quality;
+	/** Indicator if an error occurred */
+	bool err_occurred;
+	/** Unicast address*/
+	uint16_t addr;
+	/** Container of distance estimate results for a number
+	 * of different methods, in centimeters.
+	 */
+	union {
+		/** Distance estimate based on RTT measurement. */
+		uint16_t rtt;
+		struct {
+			/** Distance estimate based on IFFT of spectrum. */
+			uint16_t best;
+			/** Distance estimate based on average phase slope estimation. */
+			uint16_t ifft;
+			/** Distance estimate based on Friis path loss formula. */
+			uint16_t phase_slope;
+			/** Best effort distance estimate. */
+			uint16_t rssi_openspace;
+		} mcpd;
+	} res;
+};
+
+/** Parameters for the Distance measurement config. */
+struct bt_mesh_dm_cfg {
+	/** Default TTL value. */
+	uint8_t ttl;
+	/** Default timeout. */
+	uint8_t timeout;
+	/** Default reflector start delay. */
+	uint8_t delay;
+};
+
+/** @cond INTERNAL_HIDDEN */
+#define BT_MESH_DM_CONFIG_OP BT_MESH_MODEL_OP_3(0x01, BT_MESH_VENDOR_COMPANY_ID)
+#define BT_MESH_DM_CONFIG_STATUS_OP BT_MESH_MODEL_OP_3(0x02, BT_MESH_VENDOR_COMPANY_ID)
+#define BT_MESH_DM_START_OP BT_MESH_MODEL_OP_3(0x03, BT_MESH_VENDOR_COMPANY_ID)
+#define BT_MESH_DM_RESULT_GET_OP BT_MESH_MODEL_OP_3(0x04, BT_MESH_VENDOR_COMPANY_ID)
+#define BT_MESH_DM_RESULT_STATUS_OP BT_MESH_MODEL_OP_3(0x05, BT_MESH_VENDOR_COMPANY_ID)
+#define BT_MESH_DM_RESULT_SYNC_OP BT_MESH_MODEL_OP_3(0x06, BT_MESH_VENDOR_COMPANY_ID)
```

Comment:
How will this behave if there is a client and a server on the same element?

Reply:
I think we at least need one opcode per model unique instance

Reply:
Well, the stack will not allow one to instantiate server and client on the same element because of a single common opcode. We could split opcodes one for the client and one for the server, but it is still unnecessary IMO. We can live with the limitation that the server and client cannot be instantiated on the same element.